src/rgw/rgw_cache.h

```diff
@@ -488,7 +491,7 @@ int RGWDataCache<T>::get_obj_iterate_cb(RGWObjectCtx *ctx, RGWObjState *astate,
   if (data_cache.get(read_obj.oid)) {
     librados::L1CacheRequest *cc;
     d->add_l1_request(&cc, pbl, read_obj.oid, len, obj_ofs, read_ofs, key, c);
-    r = io_ctx.cache_aio_notifier(read_obj.oid, cc);
+    r = io_ctx.cache_aio_notifier(read_obj.oid, dynamic_cast<librados::CacheRequest*>(cc));
```

Comment:
can these dynamic_casts fail?

Reply:
changing the dynamic_cast to static_cast